apps/main/src/llamalend/features/manage-soft-liquidation/ui/tabs/ImproveHealth.tsx

```diff
@@ -61,10 +61,19 @@ export const ImproveHealth = ({
     'approve-infinite': onApproveInfinite,
   }
 
-  const maxBalance = useMemo(
+  const walletBalance = useMemo(
     () => ({
-      balance: debtToken && userBalance ? decimal(Math.min(+debtToken.amount, +userBalance).toString()) : undefined,
+      balance: userBalance,
```

Comment:
When the user has more balance than the market has liquidity, this will go wrong. I do see however that our new UX only has a place for the wallet balance, not for the max debt